client/api.go

```diff
@@ -516,15 +519,19 @@ func getContentType(appType model.AppType) string {
 	}
 }
 
-func (api API) uploadChunksParallelly(fileChunks [][]byte, chunkIDs []int, assetResponse fileAssetResponse) (retErr error) {
-	var wg sync.WaitGroup
-	wg.Add(len(fileChunks))
+func (api API) uploadChunksInParallel(fileChunks [][]byte, chunkIDs []int, assetResponse fileAssetResponse) (retErr error) {
+	sem := semaphore.NewWeighted(maxConcurrentChunkUploads)
```

Comment:
This is just a hint that I have recently limited the amount of concurrent jobs with a channel:
```
	jobs := make(chan bool, concurrency)
	for _, item := range items {
		go func(item deployment.DeployableItem) {
			jobs <- true
			...
			<-jobs
		}(item)
	}
```
I do not know what are the pros and cons of the different solution but just wanted to share it.